contracts/vaults/YearnVaults/RibbonThetaYearnVault.sol

```diff
@@ -0,0 +1,379 @@
+        depositReceipt.amount = uint104(uint256(receiptAmount).sub(amount));
+
+        emit InstantWithdraw(msg.sender, amount, currentRound);
+
+        VaultLifecycleYearn.transferAsset(
+            WETH,
+            vaultParams.asset,
+            msg.sender,
+            amount
+        );
+    }
+
+    /************************************************
+     *  VAULT OPERATIONS
+     ***********************************************/
+
+    /**
+     * @notice Sets the next option the vault will be shorting, and closes the existing short.
+     *         This allows all the users to withdraw if the next option is malicious.
+     */
+    function commitAndClose() external onlyOwner nonReentrant {
+        address oldOption = optionState.currentOption;
+
+        VaultLifecycleYearn.CloseParams memory closeParams =
+            VaultLifecycleYearn.CloseParams({
+                OTOKEN_FACTORY: OTOKEN_FACTORY,
+                USDC: USDC,
+                currentOption: oldOption,
+                delay: delay,
+                lastStrikeOverride: lastStrikeOverride,
+                overriddenStrikePrice: overriddenStrikePrice
+            });
+
+        (
+            address otokenAddress,
+            uint256 premium,
+            uint256 strikePrice,
+            uint256 delta
+        ) =
+            VaultLifecycleYearn.commitAndClose(
+                strikeSelection,
+                optionsPremiumPricer,
+                premiumDiscount,
+                closeParams,
+                vaultParams,
+                vaultState,
+                address(collateralToken)
+            );
+
+        emit NewOptionStrikeSelected(strikePrice, delta);
+
+        ShareMath.assertUint104(premium);
+        currentOtokenPremium = uint104(premium);
+        optionState.nextOption = otokenAddress;
+        optionState.nextOptionReadyAt = uint32(block.timestamp.add(delay));
+
+        _closeShort(oldOption);
+    }
+
+    /**
+     * @notice Closes the existing short position for the vault.
+     */
+    function _closeShort(address oldOption) private {
+        optionState.currentOption = address(0);
+        vaultState.lastLockedAmount = vaultState.lockedAmount;
+        vaultState.lockedAmount = 0;
+
+        if (oldOption != address(0)) {
+            uint256 withdrawAmount =
+                VaultLifecycleYearn.settleShort(GAMMA_CONTROLLER);
+            emit CloseShort(oldOption, withdrawAmount, msg.sender);
+        }
+    }
+
+    /**
+     * @notice Rolls the vault's funds into a new short position.
+     */
+    function rollToNextOption() external nonReentrant {
+        (address newOption, uint256 lockedBalance) = _rollToNextOption();
+
+        vaultState.lockedAmount = uint104(lockedBalance);
+
+        emit OpenShort(newOption, lockedBalance, msg.sender);
+
+        VaultLifecycleYearn.createShort(
+            GAMMA_CONTROLLER,
+            MARGIN_POOL,
+            newOption,
+            wdiv(
```

Comment:
We can try removing `DSMath` entirely as an import by copy pasting the wdiv/wmul functions into the contract, see if that helps with the size